### PR TITLE
feat(utxo-lib): allow both compressed & uncompressed pubkeys

### PR DESCRIPTION
**Pull request by rushilbg**
in output script creation of v1 safe wallets

Also export legacySafe functions from utxolib

BTC-335

### CODE REVIEW

File: modules/utxo-lib/test/bitgo/legacysafe/index.ts [hunk 1]
```diff
@@ -34,9 +34,9 @@ describe('legacy safe scripts creation', function () {
   const compressedPubKeys = compressedKeyPairs.map((keyPair) => keyPair.publicKey);
 
   it('throws error for compressed public keys', function () {
-    assert.throws(
+    assert.doesNotThrow(
```
Comment (OttoAllmendinger) [inline modules/utxo-lib/test/bitgo/legacysafe/index.ts:37]: the second arg here is not required anymore

you could add a test with a empty buffer or something

**Review by OttoAllmendinger — CHANGES_REQUESTED** [review]

File: modules/utxo-lib/src/bitgo/legacysafe/index.ts [hunk 2]
```diff
@@ -48,8 +48,9 @@ export function createLegacySafeOutputScript2of3(
   }
 
   pubkeys.forEach((key) => {
-    if (key.length !== 65) {
-      throw new Error(`Unexpected key length ${key.length}. Must use uncompressed keys.`);
+    // V1 Safe BTC wallets could contain both uncompressed and compressed pubkeys
```
Comment (OttoAllmendinger) [inline modules/utxo-lib/src/bitgo/legacysafe/index.ts:51]: technically I believe they must either all be compressed or uncompressed but we probably won't encounter a mixed-length scenario

**Review by OttoAllmendinger — DISMISSED** [review]

File: modules/utxo-lib/src/bitgo/legacysafe/index.ts [hunk 3]
```diff
@@ -48,8 +48,9 @@ export function createLegacySafeOutputScript2of3(
...
+    if (key.length !== 65 && key.length !== 33) {
```
Comment (OttoAllmendinger) [inline modules/utxo-lib/src/bitgo/legacysafe/index.ts:52]:
```suggestion
    if (key.length !== keys[0].length) { throw new Error(`inconsistent key lengths`); }
    if (key.length !== 65 && key.length !== 33) {
```
    if (key.length !== 65 && key.length !== 33) {
```

**Review by OttoAllmendinger — DISMISSED** [review]

File: modules/utxo-lib/test/bitgo/legacysafe/index.ts [hunk 4]
```diff
@@ -31,12 +31,11 @@ describe('public key conversion', function () {
...
   const compressedKeyPairs = getKeyTriple('utxo');
-  const compressedPubKeys = compressedKeyPairs.map((keyPair) => keyPair.publicKey);
 
-  it('throws error for compressed public keys', function () {
+  it('throws error for invalid public keys', function () {
     assert.throws(
-      () => createLegacySafeOutputScript2of3(compressedPubKeys, networks.bitcoin),
-      /^Error: Unexpected key length 33. Must use uncompressed keys.$/
+      () => createLegacySafeOutputScript2of3([Buffer.alloc(10), Buffer.alloc(11), Buffer.alloc(12)], networks.bitcoin),
+      /^Error: Unexpected key length 10, neither compressed nor uncompressed.$/
```
Comment (OttoAllmendinger) [inline modules/utxo-lib/test/bitgo/legacysafe/index.ts:38]: shouldn't this be a different error now

**Review by OttoAllmendinger — APPROVED** [review]
